cpp/memilio/compartments/compartmentalmodel.h

```diff
@@ -190,9 +155,198 @@ struct CompartmentalModel {
+    {
+        dydt.setZero();
+        for (size_t i = 0; i < this->populations.numel() / static_cast<size_t>(Comp::Count); i++) {
+            // see get_flow_index for more info on these index calculations.
+            // the Comp-index is added in get_rhs_impl; the flat Pop::Index is
+            //   flat_pop_index = i % m_comp_factor + (i / m_comp_factor) * m_comp_factor * Comp::Count;
+            // mind the integer division.
+            get_rhs_impl<0>(flows, dydt,
+                            i + (i / m_comp_factor) * m_comp_factor * (static_cast<size_t>(Comp::Count) - 1),
+                            i * FlowChart<Flows...>().size());
+        }
+    }
+
+    /**
+     * @brief compute the right-hand-side of the ODE dydt = f(y, t)
+     *
+     * This function is uses get_flow(..., flows) and get_derivatives(flows, dydt) to provide the
+     * same interface as a CompartmentalModel without flows
+     *
+     * @param pop the current population of the model as a flat array
+     * @param y the current state of the model as a flat array
+     * @param t the current time
+     * @param dydt a reference to the calculated output
+     */
+    void get_derivatives(Eigen::Ref<const Eigen::VectorXd> pop, Eigen::Ref<const Eigen::VectorXd> y, double t,
+                         Eigen::Ref<Eigen::VectorXd> dydt) const override final
+    {
+        m_flow_values.setZero();
+        get_flows(pop, y, t, m_flow_values);
+        get_derivatives(m_flow_values, dydt);
+    }
+
+    /**
+     * @brief initial values for flows.
+     * This can be used as initial conditions in an ODE solver. By default, this is a zero vector
+     * @return the initial flows
+     */
+    Eigen::VectorXd get_initial_flows() const
+    {
+        return Eigen::VectorXd::Zero((this->populations.numel() / static_cast<size_t>(Comp::Count)) *
+                                     FlowChart<Flows...>().size());
+    }
+
+    /**
+     * @brief a flat index into an array of flows (as computed by get_flows), given the indices of each category
+     * @param indices the custom indices for each category (except Comp)
+     * @tparam Source the source of a flow
+     * @tparam Target the target of a flow
+     * @return a flat index into a data structure storing flow values
+     */
+    template <Comp Source, Comp Target>
+    size_t get_flow_index(const FlowIndex& indices) const
+    {
+        // Calculation of flat indices (as in this->populations.get_flat_index) use the formula
+        // flat_index = ((((I_{1}) * D_{2} + I_{2}) * D_{3} + I_{3}) ... ) * D_{n} + I_{n}
+        //            = I_{n} + I_{n-1} * D_{n} + I_{n-2} * D_{n} * D_{n-1} + ... + I_{1} * D_{n} * ... * D_{2}
+        // for indices (I_{1}, ... , I_{n}) and dimension (D_{1}, ... , D_{n}).
+        //
+        // The flows use their position in the FlowChart instead of Comp. As this dimension (the number of flows) may
```

Comment:
Maybe the whole comment story should go in the description in doxygen (i.e. after the @brief line)